### PR TITLE
feat: create hyperctl docker image

### PR DESCRIPTION
**Pull request by minzcmu**
What each job is doing:

- `main`: Test out if the `download_hyperctl.sh` is working and if `hyperctl` command available. 
- `publish`: Download `hyperctl` binary and publish the `hyperctl` binary as a github release.

Docker build will be triggered automatically when anything new is pushed to the repository.
It will wget the `hyperctl` binary from github releases and add other scripts to the image. 
Please refer to the `Dockerfile` for details.

You can review this together with the internal repo.
Related to https://github.com/screwdriver-cd/screwdriver/issues/595

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,25 @@
+FROM alpine:3.5
+
+RUN mkdir /sd
+ADD scripts/hyper-runner.sh /sd/
+ADD scripts/install_docker.sh /sd/
+ADD scripts/hyper-pod-template.json /sd/
```
Comment (FenrirUnbound) [inline Dockerfile]: Can these three lines not be combined into `ADD scripts /sd`?

File: README.md [hunk 2]
```diff
@@ -26,6 +33,6 @@ Code licensed under the BSD 3-Clause license. See LICENSE file for terms.
 [issues-image]: https://img.shields.io/github/issues/screwdriver-cd/hyperctl-image.svg
 [issues-url]: https://github.com/screwdriver-cd/hyperctl-image/issues
 [status-image]: https://cd.screwdriver.cd/pipelines/pipelineid/badge
-[status-url]: https://cd.screwdriver.cd/pipelines/pipelineid
+[status-url]: https://cd.screwdriver.cd/pipelines/254
```
Comment (FenrirUnbound) [inline README.md]: You might want to go through and remove the badges that aren't relevant, like the NPM badge

File: screwdriver.yaml [hunk 3]
```diff
@@ -2,20 +2,28 @@ workflow:
...
+    image: centos
 
 jobs:
     main:
         steps:
-            - install: npm install
-            - test: npm test
-
+            # To verify if the hyperctl-download script works
+            - hyperctl-download: bash -xe scripts/hyperctl-download.sh
```
Comment (FenrirUnbound) [inline screwdriver.yaml:11]: Not a blocker: add a step that explicitly checks the binary exists after download

File: screwdriver.yaml [hunk 4]
```diff
@@ -2,20 +2,28 @@ workflow:
...
-            - install: npm install semantic-release
-            - publish: npm run semantic-release
+            - hyperctl-download: bash -xe scripts/hyperctl-download.sh
+            - chmod: chmod +x ./hyperctl
+            - test-hyperctl: ./hyperctl info -h
+            - setup-ci: git clone https://gist.github.com/3d2388b2a7ba658cdcdaffa8cd874e50.git ci
+            - tag: ./ci/git-tag.sh
+            - release: ./ci/git-release.sh
+        environment:
+            DOCKER_REPO: screwdrivercd/hyperctl-image
```
Comment (FenrirUnbound) [inline screwdriver.yaml]: I don't see this environment value being used in either the `tag` or `release` steps. Is it being used by the `hyperctl-download.sh`?

**Review by FenrirUnbound — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 1 above)
Comment (minzcmu) [reply Dockerfile]: Good point!

File: screwdriver.yaml (hunk 4 above)
Comment (minzcmu) [reply screwdriver.yaml]: Do not need this any more, left over from my last commit. Good catch!

File: screwdriver.yaml (hunk 3 above)
Comment (minzcmu) [reply screwdriver.yaml:11]: Next step is `chmod +x hyperctl`, if the file does not exists, chmod will throw err saying file not exist. Is that good enough?

File: Dockerfile [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+    && set -x \
+    # Alpine ships with musl instead of glibc (this fixes the symlink)
+    && mkdir /lib64 \
+    && ln -s /lib/libc.musl-x86_64.so.1 /lib64/ld-linux-x86-64.so.2 \
+    && apk add --no-cache --update ca-certificates \
+    && apk add --virtual .build-dependencies wget \
+    && apk add --update bash sudo \
+    # Download hyperctl binary
+    && wget -q -O - https://github.com/screwdriver-cd/hyperctl-image/releases/latest \
+     | egrep -o '/screwdriver-cd/hyperctl-image/releases/download/v[0-9.]*/hyperctl' \
```
Comment (dchinmaya) [inline Dockerfile]: can you add some comments here? What exactly is happening here?

**Comment by dchinmaya** [comment]
lgtm 💯

File: screwdriver.yaml (hunk 3 above)
Comment (FenrirUnbound) [reply screwdriver.yaml:11]: It's fine. It's just not explicit that it's missing.

**Review by FenrirUnbound — COMMENTED** [review]
👍 

Looks good to me

dangerousresponsiblejellyfish-size_restricted

**Review by FenrirUnbound — APPROVED** [review]

**Pull request merged**